src/domain/authentication/index.types.d.ts

```diff
@@ -0,0 +1,37 @@
+type SessionId = string & { readonly brand: unique symbol }
+type SchemaId = string & { readonly brand: unique symbol }
+
+type AuthenticationError = import("./errors").AuthenticationError
+
+type IdentityPassword = string & { readonly brand: unique symbol }
+
+type KratosUserId = string & { readonly brand: unique symbol }
+type SessionToken = string & { readonly brand: unique symbol }
+
+type Identity = {
+  id: KratosUserId
+  schemaId: SchemaId
+  traits: {
```

Comment:
I would flatten the traits here - no need to keep the structure of the `ory/client` type.